build-packages.js

```diff
@@ -37,9 +37,9 @@ fs.readdir(entryDirPath, async (e, files) => {
         main: "index.js",
         module: "index.module.js",
         esmodule: "index.esmodule.js",
-        "umd:main": "dist/index.umd.js",
+        "umd:main": "index.umd.js",
         source: `../src/packages/${file}`,
-        types: "../types/index.d.ts",
+        types: "index.d.ts",
```

Comment:
Hey @RyanChristian4427! How are you?
Do we still have issues with TS bundling or you have solve them?

Reply:
Types are definitely still messed up, Microbundle doesn't allow to target any types deeper than top level it seems.

Reply:
Any ideas on how to fix that?

If we won't include `types: "index.d.ts",` to package.json by default, it helps?

If it doesn't I could try to rewrite the build-packages.js to simulate command running inside subfolders. I guess that's not a big deal.

Reply:
We could move everything from `packages` up a level, maybe edit the build script to just target specific files rather than building the queue dynamically. Or rewrite each `package.json` after the fact with the correct paths. 

I opened an issue on the Microbundle repo just to ensure this is the case as nothing I've tried can "un-bury" the types.

Reply:
Just found out that node's exec method os also supports cwd option:
That could make Microbundle ignore the root package options and structure:
`exec('git status', {cwd: '/home/ubuntu/distro'})`

Reply:
Sorry, I'm not following, how would that help?

Microbundle builds the type definitions into whatever directory is specified in `package.json`. It assumes we have an `index.ts` at the top level and will generate the directory structure to make that happen.

Reply:
Sorry, I understood the problem wrong.
The types are wrong because all entry files are placed at the same level as the files they use, but they should be upper (more "root"), right?

Reply:
Let's just move all package entry files to /src then?
```
src/packages/hex.ts -> src/index.ts (or src/dist.ts)

src/packages/rgb.ts -> src/rgb.ts

...
```

Reply:
Yeah, pretty much. 

It's a bit difficult to describe over text, but if you clone and build locally it becomes pretty clear. 

Microbundle will toss all children of `src` into the top-most directory specified in the types field of` a `packages.json`. So if we have `"types": "dist/index.d.ts"`, our output would be:

```
dist/
--components/
--packages/
--utils/
...
```

If we target lower, like `"types": "dist/packages/hex.d.ts"` we get:

```
dist/
--packages/
----components/
----packages/
----utils
...
```

Quite difficult.

Reply:
> Let's just move all package entry files to /src then?
> 
> ```
> src/packages/hex.ts -> src/index.ts (or src/dist.ts)
> 
> src/packages/rgb.ts -> src/rgb.ts
> 
> ...
> ```

We certainly could do. Just need to change the build script to be more explicit in we want built then.

I was attempting to see if there was some way we could keep the current structure but if this works for you then great.

Reply:
Yeah, it works) If we'll find a way to keep the current structure, we could do it a a separate PR. Let's go with moving eveything to /src since we know that it solves everything and allows us to finish this task and merge the changes.

Reply:
Thanks again for your patience. It took a while to understand the problem) My bad)

Reply:
Oh this is quite awkward, I'm sure I didn't explain it well. Trying to type out directory structure in text form is never easy.